core/src/auctions.ts

```diff
@@ -67,26 +67,50 @@ const calculateCollateralToCoverDebt = async function (network: string, auction:
     return auction.debtDAI.dividedBy(marketPriceForAllCollateral);
 };
 
-const enrichAuctionWithMarketValues = async function (auction: Auction, network: string): Promise<Auction> {
+export const enrichAuctionWithMarketData = async function (auction: Auction, network: string): Promise<Auction> {
     if (!auction.isActive || !auction.approximateUnitPrice || auction.isFinished) {
         return auction;
     }
     try {
         const collateralToCoverDebt = await calculateCollateralToCoverDebt(network, auction);
-        const marketUnitPrice = await getMarketPrice(network, auction.collateralSymbol, collateralToCoverDebt);
-        const marketUnitPriceToUnitPriceRatio = auction.approximateUnitPrice
-            .minus(marketUnitPrice)
-            .dividedBy(marketUnitPrice);
-        const auctionWithMarketValues = {
+        let marketDataRecords = await getMarketData(network, auction.collateralSymbol, collateralToCoverDebt);
+        const exchangeFees = await getDefaultMarketFee(network);
+        const exchangeFeePerUnitDAI = exchangeFees.exchangeFeeDAI.dividedBy(collateralToCoverDebt);
+        const currentDate = await getNetworkDate(auction.network);
+        for (let marketData of Object.values(marketDataRecords)) {
+            marketData = {
+                ...marketData,
+                marketUnitPrice: marketData.marketUnitPrice.plus(exchangeFeePerUnitDAI),
+                marketUnitPriceToUnitPriceRatio: auction.approximateUnitPrice
+                    .minus(marketData.marketUnitPrice)
+                    .dividedBy(marketData.marketUnitPrice),
+                ...exchangeFees,
+                transactionGrossProfit: calculateTransactionGrossProfit(
+                    marketData.marketUnitPrice,
+                    collateralToCoverDebt,
+                    auction.approximateUnitPrice
+                ),
+                transactionGrossProfitDate: calculateTransactionGrossProfitDate(
+                    auction,
+                    marketData.marketUnitPrice,
+                    currentDate
+                ),
+            };
```

Comment:
The idea of the last suggestion was to move this whole block with calculations into the `getMarketDataById`. The reason is: it's extremely confusing that everything is a `marketData` and there is no types to enforce proper keys. You call `makertData` `collateral.exchanges[marketId]` in one place, `{ errorMessage: error, marketUnitPrice: new BigNumber(NaN), route: [], }` in another, then here you call `marketData` something else, but those are all different thing and we have typescript to enforce them